subscription/src/main/java/org/killbill/billing/subscription/api/svcs/DefaultSubscriptionInternalApi.java

```diff
@@ -211,66 +212,120 @@ public SubscriptionBase createSubscription(final UUID bundleId, final PlanPhaseS
-                subscription.setInitialPhase(spec.getPhaseType());
-                subscription.setBuilder(new SubscriptionBuilder()
-                                                .setId(UUIDs.randomUUID())
-                                                .setBundleId(bundleId)
-                                                .setCategory(plan.getProduct().getCategory())
-                                                .setBundleStartDate(effectiveDate)
-                                                .setAlignStartDate(effectiveDate)
-                                                .setMigrated(isMigrated));
-
-                subscriptions.add(subscription);
+            SubscriptionSpecifier subscription = new SubscriptionSpecifier();
+            subscription.setRealPriceList(plan.getPriceListName());
+            subscription.setEffectiveDate(effectiveDate);
+            subscription.setProcessedDate(now);
+            subscription.setPlan(plan);
+            subscription.setInitialPhase(spec.getPhaseType());
+            subscription.setBuilder(new SubscriptionBuilder()
+                                            .setId(UUIDs.randomUUID())
+                                            .setBundleId(bundleId)
+                                            .setCategory(plan.getProduct().getCategory())
+                                            .setBundleStartDate(effectiveDate)
+                                            .setAlignStartDate(effectiveDate)
+                                            .setMigrated(isMigrated));
+
+            subscriptions.add(subscription);
+        }
+        return subscriptions;
+    }
+
+    @Override
+    public List<SubscriptionBase> createBaseSubscriptionsWithAddOns(final UUID accountId, final Iterable<BaseEntitlementWithAddOnsSpecifier> baseEntitlementWithAddOnsSpecifier, final InternalCallContext context) throws SubscriptionBaseApiException {
+        try {
+            final Catalog catalog = catalogService.getFullCatalog(true, true, context);
+            final CallContext callContext = internalCallContextFactory.createCallContext(context);
+
+            final List<SubscriptionAndAddOnsSpecifier> subscriptionAndAddOns = new ArrayList<SubscriptionAndAddOnsSpecifier>();
+            for (BaseEntitlementWithAddOnsSpecifier entitlementWithAddOnsSpecifier : baseEntitlementWithAddOnsSpecifier) {
+                final DateTime now = clock.getUTCNow();
+                final DateTime effectiveDate = (entitlementWithAddOnsSpecifier.getEntitlementEffectiveDate() != null) ?
+                                               DefaultClock.truncateMs(entitlementWithAddOnsSpecifier.getEntitlementEffectiveDate().toDateTimeAtCurrentTime()) : now;
+
+                final SubscriptionBaseBundle bundle = createBundleForAccount(accountId, entitlementWithAddOnsSpecifier.getExternalKey(), context);
+
+                final List<SubscriptionSpecifier> subscriptions = verifyAndBuildSubscriptionSpecifiers(bundle.getId(),
```

Comment:
> I think that's not enough, since the getSubscriptionsForBundle() looks for the existing Subscriptions (already created in the DB) and then it is being verified against the ones that are about to be added.

In that case i think this is enough because you know there is nothing on disk (there are no existing subscriptions, since all the bundles created in the shopping card api were new? Or is that assumption wrong? Do we support to both create new bundles/subscriptions and also add subscriptions (AO) to existing bundles?